geoh5py/ui_json/forms.py

```diff
@@ -453,23 +493,34 @@ def __init__(self, name, value=None, **kwargs):
             ],
         )
         self._is_value = BoolParameter("is_value")
-        self._property = UUIDParameter("property")
+        self._property = TypeRestrictedParameter(
+            "property", [self._data_type_string_to_type(data_type)]
+        )
         value = NumericParameter("value", value=value)
-        super().__init__(name, value=value, **kwargs)
+        super().__init__(name, value=value, data_type=data_type, **kwargs)
 
     @property
     def value(self):
         """Form value is value of property when is_value is False."""
-        val = self.property
-        if self.is_value:
+        val = self.property  # type: ignore # pylint: disable=no-member
+        if self.is_value:  # type: ignore # pylint: disable=no-member
             val = self._value.value
         return val
 
     @value.setter
     def value(self, val):
         if isinstance(val, (int, float)):
             self._value.value = val
-            self.is_value = True
         else:
-            self.property = val
-            self.is_value = False
+            self._property.value = val
+
+    def _data_type_string_to_type(self, data_type: str) -> type:

```

Comment:
This could be static, even better make it a constant. You will have to check that the key is valid wherever you are calling this.

Reply:
Done.  Note, I use .get with a None default to the lookup so I don't have to check for a valid key.